### PR TITLE
MGMT-8192: Logs_cmd command fails if insecure flag is set to true

### PR DESCRIPTION
**Pull request by tsorya**
## Description
[MGMT-8192](https://issues.redhat.com/browse/MGMT-8192): Logs_cmd command fails if insecure flag is set to true 
When insecure=true logs sender will try to use certificate that was set in --cacert flag.
Service currently always sets "cacert" flag with path that was not mounted and not exist.

This commit fixes setting cacert flag all the time and will set it only
in case certificate path was provided, same as in install_cmd

## List all the issues related to this PR

- [ ] New Feature
- [X] Bug fix
- [X] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [X] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @oshercc 
/cc 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/hostcommands/logs_cmd.go [hunk 1]
```diff
@@ -83,10 +83,15 @@ func createUploadLogsCmd(host *models.Host, baseURL, agentImage, mastersIPs stri
 		"BOOTSTRAP":              strconv.FormatBool(host.Bootstrap),
 		"INSTALLER_GATHER":       strconv.FormatBool(withInstallerGatherLogging),
 		"MASTERS_IPS":            mastersIPs,
-		"CACERTPATH":             common.HostCACertPath,
 	}
+
+	if i.instructionConfig.ServiceCACertPath != "" {
```
Comment (omertuc) [inline internal/host/hostcommands/logs_cmd.go:88]: Isn't the `{{if .CACERTPATH}}` inside the template enough? Why do we need this `if` as well?

Comment (omertuc) [reply internal/host/hostcommands/logs_cmd.go:88]: Oh nevermind ignore me

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/test e2e-metal-assisted-operator-ztp
seems like a flake

**Comment by tsorya** [comment]
/retitle Bug 2024156: Logs_cmd command fails if insecure flag is set to true

**Pull request merged**